h2o-algos/src/main/java/hex/ensemble/Metalearners.java

```diff
@@ -100,23 +103,91 @@ protected SimpleMetalearner(String algo) {
         ModelBuilder createBuilder() {
             return ModelBuilder.make(_algo, _metalearnerJob, _metalearnerKey);
         }
+
+        protected String getAlgo() {
+            return _algo;
+        }
+    }
+
+    static class MetalearnerWithDistribution extends SimpleMetalearner {
+        protected DistributionFamily[] _supportedDistributionFamilies;
+        protected MetalearnerWithDistribution(String algo) {
+            super(algo);
+        }
+        @Override
+        protected void validateParams(Model.Parameters parms) {
+            super.validateParams(parms);
+
+            // Check if distribution family is supported and if not pick a basic one
+            if (!ArrayUtils.contains(_supportedDistributionFamilies, parms._distribution)) {
+                DistributionFamily distribution;
+                if (_model._output.nclasses() == 1) {
+                    distribution = DistributionFamily.gaussian;
+                } else if (_model._output.nclasses() == 2) {
+                    distribution = DistributionFamily.bernoulli;
+                } else {
+                    distribution = DistributionFamily.multinomial;
+                }
+
+                Log.warn("Distribution \"" + parms._distribution +
+                        "\" is not supported by metalearner algorithm \"" + getAlgo() +
+                        "\". Using \"" + distribution + "\" instead.");
+
+                parms._distribution = distribution;
+            }
+            _model._dist = DistributionFactory.getDistribution(parms);
```

Comment:
is it possible to move this last statement in `compute()`?
I'm not a fan of validating params and modifying the model in the same place.

Reply:
you could add another hook to `Metalearner` like you did with `validateParams`

Reply:
I moved it to `compute` but instead of creating it by `DistributionFactory` I use the same `_dist` as the metalearner which I think is more semantically correct. And if the metalearner model such as GLM ignores it, SE should ignore it as well I think and since the `_dist` is property of `Model.Parameters` it should be safe to do it this way and potentially even safer, e.g., if the underlying model coerces the family/distribution somehow during its creation.